modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/Parser.java

```diff
@@ -503,16 +510,29 @@ protected SAXParser parser(boolean validate) throws ParserConfigurationException
         // add the handler as a LexicalHandler too.
         parser.setProperty(SAX_PROPERTY_PREFIX + LEXICAL_HANDLER_PROPERTY, handler);
         // set Entity expansion limit
-        parser.setProperty(
-                JDK_ENTITY_EXPANSION_LIMIT,
-                entityExpansionLimit != null
-                        ? entityExpansionLimit
-                        : DEFAULT_ENTITY_EXPANSION_LIMIT);
+        setupEntityExpansionLimit(parser);
+
         //
         // return builded parser
         return parser;
     }
 
+    private void setupEntityExpansionLimit(final SAXParser parser) throws SAXNotSupportedException {
+        try {
+            parser.setProperty(
+                    JDK_ENTITY_EXPANSION_LIMIT,
+                    entityExpansionLimit != null
+                            ? entityExpansionLimit
+                            : DEFAULT_ENTITY_EXPANSION_LIMIT);
+        } catch (SAXNotRecognizedException ex) {
+            LOGGER.warning(
+                    "Sax parser property '"
+                            + JDK_ENTITY_EXPANSION_LIMIT
+                            + "' not recognized.  "
+                            + "Please check if XercesImpl jar dependency must be updated.");
```

Comment:
Could actually be newer and require a different property. Better to just say that the xerces version is incompatible.